src/components/bc-balance.ts

```diff
@@ -85,14 +85,18 @@ export class Balance extends withTwind()(BitcoinConnectElement) {
         if (!provider) {
           return;
         }
-        // TODO: consider using getInfo to check if balance is supported
-        // and do not show an error if it is not supported
-        // (do not display the balance - needs to be handled somewhere else)
-        if (!provider.getBalance) {
+
+        const info = store.getState().info;
+        if (
+          !info?.methods ||
+          info.methods.indexOf('getBalance') < 0 ||
+          !provider.getBalance
```

Comment:
to me these checks are always a bit of a code-smell. and `if` is one already and then the more complicated it is the more problematic it is to read it easily and understand what is going for a new developer.

in this PR you do this check already at 3 places that would need to be kept in sync. 
be careful there and we need to think about the clear defined API (internal and external)

Reply:
I agree. Maybe we calculate this once when the connection is initialized and remove the check from the balance component (it should never reach that). I'll try clean this up tomorrow.